src/main/java/games/strategy/triplea/delegate/MustFightBattle.java

```diff
@@ -2550,15 +2552,17 @@ private void checkDefendingPlanesCanLand() {
   }
 
   static CompositeChange clearTransportedByForAlliedAirOnCarrier(final Collection<Unit> attackingUnits,
-      final PlayerID attacker, final GameData data) {
+      final Territory battleSite, final PlayerID attacker, final GameData data) {
     final CompositeChange change = new CompositeChange();
     // Clear the transported_by for successfully won battles where there was an allied air unit held as cargo by an
     // carrier unit
     final Collection<Unit> carriers = Matches.getMatches(attackingUnits, Matches.unitIsCarrier());
     if (!carriers.isEmpty() && !Properties.getAlliedAirIndependent(data)) {
       final Predicate<Unit> alliedFighters = Match.allOf(Matches.isUnitAllied(attacker, data),
           Matches.unitIsOwnedBy(attacker).negate(), Matches.unitIsAir(), Matches.unitCanLandOnCarrier());
-      final Collection<Unit> alliedAirInTerr = Matches.getMatches(attackingUnits, alliedFighters);
+      final Collection<Unit> alliedAirInTerr = Matches.getMatches(
+          CollectionUtils.union(attackingUnits, battleSite.getUnits()),
+          alliedFighters);
```

Comment:
Would it be better to just use Java 8 here or is there a guava equivalent since we already include that?

Reply:
I considered using Guava's `Sets.union()`, but it required converting the existing `Collection`s to `Set`s, which seemed a bit noisy:

```java
final Collection<Unit> alliedAirInTerr = Matches.getMatches(
    Sets.union(Sets.newHashSet(attackingUnits), Sets.newHashSet(battleSite.getUnits())),
    alliedFighters);
```

As I said above, Java 8 streams can also be used, but the `toSet()` call could be easily overlooked, and I didn't want to add the comment `// remove duplicates`.

I can understand if the concern is the introduction of a new dependency.  I guess the Guava alternative is more in-your-face "this is removing duplicates" because the word "set" appears no less than five times. :smile:

Reply:
Yeah, I glanced at the Guava alternative and realized the same thing. I tend to lean on limiting new dependencies as it makes it harder for new developers.